spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -2331,17 +2400,18 @@ public void testMergeSinglePartitionPartitioning() {
         "MERGE INTO %s t USING source s ON t.id = s.id "
             + "WHEN MATCHED THEN UPDATE SET *"
             + "WHEN NOT MATCHED THEN INSERT *",
-        tableName);
+        commitTarget());
 
     ImmutableList<Object[]> expectedRows =
         ImmutableList.of(row(-1), row(0), row(1), row(2), row(3), row(4));
 
-    List<Object[]> result = sql("SELECT * FROM %s ORDER BY id", tableName);
+    List<Object[]> result = sql("SELECT * FROM %s ORDER BY id", selectTarget());
     assertEquals("Should correctly add the non-matching rows", expectedRows, result);
   }
 
   @Test
   public void testMergeEmptyTable() {
+    Assume.assumeFalse("Custom branch does not exist for empty table", "test".equals(branch));
```

Comment:
We will not test custom branch for empty table cases, but main will work as expected.

Reply:
We can always add this later. I'm good with not allowing branches for now.